WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverFragment.java

```diff
@@ -0,0 +1,226 @@
+    private static final String ARG_LAST_USED_BLOG_LOCAL_ID = "ARG_LAST_USED_BLOG_LOCAL_ID";
+    private static final String ARG_AFTER_LOGIN = "ARG_AFTER_LOGIN";
+
+    @Inject AccountStore mAccountStore;
+    private ShareIntentFragmentListener mShareIntentFragmentListener;
+    private SitePickerAdapter mAdapter;
+
+    private Button mSharePostBtn;
+    private Button mShareMediaBtn;
+
+    private boolean mSharingMediaFile;
+    private int mLastUsedBlogLocalId;
+    private boolean mAfterLogin;
+    private RecyclerView mRecyclerView;
+    private View mBottomButtonsContainer;
+    private View mBottomButtonsShadow;
+
+    public static ShareIntentReceiverFragment newInstance(boolean sharingMediaFile, int lastUsedBlogLocalId,
+        boolean afterLogin) {
+        ShareIntentReceiverFragment fragment = new ShareIntentReceiverFragment();
+        Bundle args = new Bundle();
+        args.putBoolean(ARG_SHARING_MEDIA, sharingMediaFile);
+        args.putBoolean(ARG_AFTER_LOGIN, afterLogin);
+        args.putInt(ARG_LAST_USED_BLOG_LOCAL_ID, lastUsedBlogLocalId);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof ShareIntentFragmentListener) {
+            mShareIntentFragmentListener = (ShareIntentFragmentListener) context;
+        } else {
+            throw new RuntimeException("The parent activity doesn't implement ShareIntentFragmentListener.");
+        }
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+        @Nullable Bundle savedInstanceState) {
+        ViewGroup layout = (ViewGroup) inflater.inflate(R.layout.login_epilogue_screen, container, false);
+        initButtonsContainer(layout);
+        initShareActionPostButton(layout);
+        initShareActionMediaButton(layout, mSharingMediaFile);
+        initRecyclerView(layout);
+        return layout;
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((WordPress) getActivity().getApplication()).component().inject(this);
+        mSharingMediaFile = getArguments().getBoolean(ARG_SHARING_MEDIA);
+        mAfterLogin = getArguments().getBoolean(ARG_AFTER_LOGIN);
+        mLastUsedBlogLocalId = getArguments().getInt(ARG_LAST_USED_BLOG_LOCAL_ID);
+        loadSavedState(savedInstanceState);
+    }
+
+    private void loadSavedState(Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            mLastUsedBlogLocalId = savedInstanceState.getInt(ARG_LAST_USED_BLOG_LOCAL_ID);
+        }
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        int selectedItemLocalId = mAdapter.getSelectedItemLocalId();
+        if (selectedItemLocalId != 1) {
```

Comment:
Should the conditional be against `-1` perhaps?

Reply:
:) Good job! Thanks